app/iot/service.py

```diff
@@ -26,24 +27,23 @@ class IOTService:
     def __init__(self) -> None:
         self.devices: dict[str, Device] = {}
 
-    def register_device(self, device: Device) -> str:
-        device.connect()
+    async def register_device(self, device: Device) -> str:
+        await device.connect()
```

Comment:
Ensure that the `connect` method in the `Device` protocol is asynchronous, as it is awaited here. If it's not defined as `async`, this will cause an error.